### PR TITLE
HYDRA-1054 : Support material assignment on a usd prim

### PR DESCRIPTION
**Pull request by lanierd-adsk**
- Support material assignment on a usd prim
- Support default material for usd procedural prims (sphere, capsule, cylinder etc..)

### CODE REVIEW

**Comment by lanierd-adsk** [comment]
FYI : @ppt-adsk @lilike-adsk

File: lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp [hunk 1]
```diff
@@ -68,7 +88,7 @@ HdSceneIndexPrim DefaultMaterialSceneIndex::GetPrim(const SdfPath& primPath) con
 bool DefaultMaterialSceneIndex::_ShouldWeApplyTheDefaultMaterial(const HdSceneIndexPrim& prim)const
 {
     // Only for meshes so far
```
Comment (debloip-adsk) [inline lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp]: Comment needs to be adjusted

**Review by debloip-adsk — CHANGES_REQUESTED** [review]
Looks good! Just one outdated comment

**Review by debloip-adsk — DISMISSED** [review]

File: lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp [hunk 2]
```diff
@@ -22,13 +22,33 @@
...
 
 PXR_NAMESPACE_USING_DIRECTIVE
 
 namespace{
     static const TfToken purposes[] = { HdMaterialBindingsSchemaTokens->allPurpose };
+
+    // Support implicit surfaces from USD as well, not only meshes
+    bool _IsDefaultMaterialCompliantPrimitive(const TfToken& primType) 
+    { 
+        static VtArray<TfToken> const compliantPrimitives = { HdPrimTypeTokens->cone,
```
Comment (ppt-adsk) [inline lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp]: Minor: would be slightly better as an std::set, with the check using either std::set::find() or std::set::count().

File: lib/mayaHydra/hydraExtensions/mixedUtils.cpp [hunk 3]
```diff
@@ -78,6 +78,11 @@ SdfPath DagPathToSdfPath(
     const bool      stripNamespaces)
 {
     std::string name = dagPath.fullPathName().asChar();
+    if ( name.empty() ) {
+        MFnDependencyNode dep(dagPath.node());
+        TF_WARN("Empty fullpath name for DAG object : %s of type : %s", dep.name().asChar(), dep.typeName().asChar());
```
Comment (ppt-adsk) [inline lib/mayaHydra/hydraExtensions/mixedUtils.cpp:83]: Do we know when this happens?  Is this for non-Dag nodes?  Does this indicate missing functionality, missing logic, or a bug?

**Review by ppt-adsk — COMMENTED** [review]
Looks good, thanks!  Only minor comments.

File: lib/mayaHydra/hydraExtensions/mixedUtils.cpp (hunk 3 above)
Comment (lanierd-adsk) [reply lib/mayaHydra/hydraExtensions/mixedUtils.cpp:83]: I had this with Bifrost nodes and also when selecting the default material set. But the problem is that wer get dag nodes with no type and no name, we should try detect useful dag nodes that need to be translated as some may not need to be treated.

File: lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp [hunk 4]
```diff
@@ -22,13 +22,33 @@
...
+                                                              HdPrimTypeTokens->cube,
+                                                              HdPrimTypeTokens->sphere,   
+                                                              HdPrimTypeTokens->capsule,
+                                                              HdPrimTypeTokens->capsule_1,
+                                                        #if HD_API_VERSION >= 67 // USD 24.05+
+                                                              UsdGeomTokens->TetMesh,
+                                                        #endif
+                                                              UsdGeomTokens->Plane,
+                                                              HdPrimTypeTokens->mesh};
+        return std::find(compliantPrimitives.cbegin(), compliantPrimitives.cend(), primType)
```
Comment (ppt-adsk) [inline lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp:49]: Must use std::set::find() here to get better performance, and it also makes the code more compact.  I also think std::set::count() is even more compact, but that's personal preference and std::set::find() is fine.

File: lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp [hunk 5]
```diff
@@ -22,13 +22,33 @@
...
+            != compliantPrimitives.cend();
```
Comment (ppt-adsk) [inline lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp]: Must use std::set::find() here to get better performance, and it also makes the code more compact. I also think std::set::count() is even more compact, but that's personal preference and std::set::find() is fine
```suggestion
        return compliantPrimitives.count(primType) == 1;
```

File: lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp
Comment (ppt-adsk) [inline lib/flowViewport/sceneIndex/fvpDefaultMaterialSceneIndex.cpp:1]: Github is erroring out saying that "You need to add a comment indicating the requested change.", even though I've done exactly that.  Hoping that this additional comment will help.

**Review by ppt-adsk — CHANGES_REQUESTED** [review]

**Comment by ppt-adsk** [comment]
Here's a useful reference:
https://stackoverflow.com/questions/17256001/merits-of-stdfind

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**